### PR TITLE
Add gmm demux

### PR DESCRIPTION
**Pull request by mari-ga**
Describe your pull request here:

_GMM-Demux is a Gaussian-Mixture-Model-based software for processing sample barcoding data (cell hashing and MULTI-seq). GMM-Demux identifies Multi-Sample Multiplets (MSMs) in a sample barcoding dataset.
Package from: https://pypi.org/project/GMM-Demux/
Repository: https://github.com/CHPGenetics/GMM-Demux_
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Review by mari-ga — COMMENTED** [review]
I included your changes, but sadly, I'm still getting the same error: conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("python==3.12.0rc3=rc3_hab00c5b_1_cpython"), MatchSpec("_python_rc")}

**Comment by aliciaaevans** [comment]
@mari-ga Looks like the conda-forge statistics package requires python <3. Python 3 has statistics built in, and argparse is also built in. So my suggestion is to remove those two from the requirements and also remove the python pinning and just have `- python` which will use the Bioconda default pinning.

File: recipes/gmm-demux/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - GMM-demux = GMM_Demux.GMM_Demux:main
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vvv --no-deps --no-build-isolation
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('gmm-demux', max_pin="x.x.x") }}
+
+requirements:
+  host:
+    - python >=3.7
```
Comment (aliciaaevans) [inline recipes/gmm-demux/meta.yaml:23]:
```suggestion
    - python
```

File: recipes/gmm-demux/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - pip
+  run:
+    - python >=3.7
```
Comment (aliciaaevans) [inline recipes/gmm-demux/meta.yaml:26]:
```suggestion
    - python
```

File: recipes/gmm-demux/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - pandas
+    - {{ pin_compatible('numpy') }}
+    - scipy
+    - tabulate
+    - argparse
+    - statistics
```
Comment (aliciaaevans) [inline recipes/gmm-demux/meta.yaml:32]:
```suggestion
```

**Review by mencian — APPROVED** [review]

**Pull request merged**